src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+const getTodoId = (todos: Omit<Todo, 'user'>[]) => {
+  return todos.reduce((maxId, todo) => Math.max(maxId, todo.id), 0);
+};
+
+const preparedTodo = (todos: Omit<Todo, 'user'>[]) => {
+  return todos.map((todo) => {
+    const findUser = usersFromServer.find((user) => user.id === todo.userId);
+
+    return { ...todo, user: findUser || null };
+  });
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isValidTitle, setIsValidTitle] = useState(true);
+  const [isValidUser, setIsValidUser] = useState(true);
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const isFormValid = title.length && userId;
+
+    if (isFormValid) {
+      setTodos((prevTodos) => {
+        const newTodo = {
+          id: getTodoId(prevTodos) + 1,
+          title,
+          completed: false,
+          userId,
+        };
+
+        return [...prevTodos, newTodo];
+      });
+      setTitle('');
+      setUserId(0);
+    } else {
+      setIsValidTitle(Boolean(title.length));
+      setIsValidUser(Boolean(userId));
+    }
+  };
+
+  const handleChangeTitle = (
+    { target: { value } }: ChangeEvent<HTMLInputElement>,
+  ) => {
```

Comment:
just add type to the event
```suggestion
  const handleChangeTitle = (event: ...) => {
```